assets/src/edit-story/app/media/useMediaReducer/reducer.js

```diff
@@ -53,36 +60,43 @@ function reducer(state, { type, payload }) {
     case types.FETCH_MEDIA_ERROR: {
       return {
         ...state,
-        media: [],
         isMediaLoaded: true,
         isMediaLoading: false,
       };
     }
 
     case types.RESET_FILTERS: {
-      return {
-        ...state,
-        searchTerm: '',
-        mediaType: '',
-      };
+      return { ...INITIAL_STATE };
     }
 
     case types.SET_SEARCH_TERM: {
       const { searchTerm } = payload;
+      if (searchTerm === state.searchTerm) return state;
       return {
-        ...state,
+        ...INITIAL_STATE,
+        mediaType: state.mediaType,
         searchTerm,
       };
     }
 
     case types.SET_MEDIA_TYPE: {
       const { mediaType } = payload;
+      if (mediaType === state.mediaType) return state;
       return {
-        ...state,
+        ...INITIAL_STATE,
+        searchTerm: state.searchTerm,
         mediaType,
       };
     }
 
+    case types.SET_PAGE: {
```

Comment:
Pls clarify: what does the `SET_PAGE` do comparing to the `FETCH_MEDIA_SUCCESS`? Is this request-vs-response data? That'd seem dangerous to conflate them into a single property, no?

Reply:
We now just have `SET_NEXT_PAGE`.